app/models/project/Project.scala

```diff
@@ -166,9 +163,15 @@ class ProjectDAO @Inject()(sqlClient: SQLClient)(implicit ec: ExecutionContext)
                           where _id = ${p._id.id}""")
     } yield ()
 
-  def updatePaused(id: ObjectId, isPaused: Boolean)(implicit ctx: DBAccessContext) =
+  def updatePaused(id: ObjectId, isPaused: Boolean)(implicit ctx: DBAccessContext): Fox[Unit] =
     updateBooleanCol(id, _.paused, isPaused)
 
+  def countForTeam(teamId: ObjectId): Fox[Int] =
```

Comment:
The `countForTeam` seem pretty WET. Maybe we can extract them to a trait or in a utils function.

Reply:
I agree that this duplication is not pretty.

However, I’m afraid my ideas of how to extract them do not really make it better
(1) move `coutForTeam` up as is to SecuredSQLDAO
 → problem: not every table has a team column
(2) move up `count` to SecuredSQLDAO, but pass the query
 → would not reduce duplication by a lot, all three would still have their query
(3) move up `count` to SecuredSQLDAO and pass only where clause as string
 → Would mean that we have to construct where clause as string, as opposed to `sql""` literal
(4) extract `countForTeam` to separate trait to have the three DAOs extend
 → that new trait would have to either extend abstract class SecuredSQLDAO itself (with all that is required there, I couldn’t get it to work) or at least declare both `run` and `existingCollectionName` with their full signatures, which is quite a bit of duplication in its own right

I think the current version is actually preferrable, despite the duplication, as it is the easiest to read. As an aside, I somewhat dislike extracting it also because it is kind of random that those tables have a `_team` column and the responsibility for knowing the columns should be with the DAOs. One may want to change this query for only one of the three in the future.
Do you think this is fair?

Reply:
Yes, I agree. This seems fair :slightly_smiling_face: